easybuild/easyblocks/i/impi.py

```diff
@@ -313,4 +314,11 @@ def make_module_extra(self, *args, **kwargs):
             # -fc also works, but -f90 takes precedence
             txt += self.module_generator.set_alias('mpiifort', 'mpiifort -f90=ifort')
 
+        # set environment variable UCX_TLS to 'all', this will work in all setups
+        # needed with UCX regardless of the transports available (even without a Mellanox HCA)
+        # since impi v2019.8, the MLX provider works without UCX_TLS, but setting it does not hurt
+        ucx_root = get_software_root('UCX')
+        if ucx_root:
+            txt += self.module_generator.set_environment('UCX_TLS', 'all')
```

Comment:
Are you sure that all versions of UCX up until 1.9.0 knows about "all" ?

Reply:
yes, added ages ago with https://github.com/openucx/ucx/pull/104

Reply:
ok

Reply:
> also what is the difference between UCX_TLS=all and not setting UCX_TLS? Does UCX_TLS=all include more transports?

There is no difference, in both cases UCX will consider all available TLS and choose the best one. So explicitly setting `UCX_TLS=all` will not change the behaviour of systems that were already working well without it.